index.js

```diff
@@ -612,26 +621,24 @@ HardSourceWebpackPlugin.prototype.apply = function(compiler) {
                 return carry;
               }, {});
             }
-            if (!HardModule.needRebuild(
-              cacheItem.buildTimestamp,
-              cacheItem.fileDependencies,
-              cacheItem.contextDependencies,
-              // [],
-              fileTimestamps,
-              compiler.contextTimestamps
-            )) {
-              var module = new HardModule(cacheItem);
-
-              // Custom plugin handling for common plugins.
-              // This will be moved in a pluginified HardSourcePlugin.
-              if (cacheItem.extractTextPluginMeta) {
-                module[extractTextNS] = cacheItem.extractTextPluginMeta;
-              }
 
-              return cb(null, module);
-            }
+            promise = md5(result.resource).then(function(md5Contents) {
```

Comment:
I see - I was doing this in the wrong place.

I've added the md5 checks to be at the same time as the mtime recording, except in the cases where there is no dataCache, in which case I do it when we create the module cache. I've added the md5 check to needRebuild as additional parameters